akka-docs/src/main/paradox/typed/persistence-durable-state.md

```diff
@@ -0,0 +1,391 @@
+* `thenReply` send a reply message to the given `ActorRef`
+
+In the example below, we use a different constructor of `DurableStateBehavior.withEnforcedReplies`, which creates
+a `Behavior` for a persistent actor that ensures that replies to commands are not forgotten. Hence it will be
+a compilation error if the returned effect isn't a `ReplyEffect`.
+
+Instead of `Increment` we will have a new command `IncrementWithConfirmation` that, along with persistence will also
+send an acknowledgement as a reply to the `ActorRef` passed in the command. 
+
+Example of effects and side-effects:
+
+Scala
+:  @@snip [DurableStatePersistentBehaviorCompileOnly.scala](/akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/DurableStatePersistentBehaviorCompileOnly.scala) { #effects }
+
+Java
+:  @@snip [DurableStatePersistentBehaviorTest.java](/akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/DurableStatePersistentBehaviorTest.java) { #effects }
+
+Most of the time this will be done with the `thenRun` method on the `Effect` above. You can factor out
+common side effects into functions and reuse for several commands. For example:
+
+Scala
+:  @@snip [PersistentActorCompileOnlyTest.scala](/akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/PersistentActorCompileOnlyTest.scala) { #commonChainedEffects }
+
+Java
+:  @@snip [PersistentActorCompileOnlyTest.java](/akka-persistence-typed/src/test/java/akka/persistence/typed/javadsl/PersistentActorCompileOnlyTest.java) { #commonChainedEffects }
+
+### Side effects ordering and guarantees
+
+Any side effects are executed on an at-most-once basis and will not be executed if the persist fails.
+
+Side effects are not run when the actor is restarted or started again after being stopped.
+
+The side effects are executed sequentially, it is not possible to execute side effects in parallel, unless they
+call out to something that is running concurrently (for example sending a message to another actor).
+
+It's possible to execute a side effect before persisting the state, but that can result in that the
+side effect is performed but the state is not stored if the persist fails.
+
+## Cluster Sharding and DurableStateBehavior
+
+In a use case where the number of persistent actors needed is higher than what would fit in the memory of one node or
+where resilience is important so that if a node crashes the persistent actors are quickly started on a new node and can
+resume operations @ref:[Cluster Sharding](cluster-sharding.md) is an excellent fit to spread persistent actors over a
+cluster and address them by id.
+
+The `DurableStateBehavior` can then be run as with any plain actor as described in @ref:[actors documentation](actors.md),
+but since Akka Persistence is based on the single-writer principle the persistent actors are typically used together
+with Cluster Sharding. For a particular `persistenceId` only one persistent actor instance should be active at one time.
```

Comment:
mixing entity with actor, though entity is not explained. Maybe just say "ensures that there is only one active actor instance for each id' ?

Reply:
done ..